scene/gui/line_edit.cpp

```diff
@@ -268,7 +268,9 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 				return;
 			}
 
-			shift_selection_check_pre(b->is_shift_pressed());
+			if (b->is_shift_pressed()) {
+				shift_selection_check_pre(b->is_shift_pressed());
+			}
```

Comment:
What is this change about?
Also if anything, you should do `shift_selection_check_pre(true)`, because you are checking it in the condition before.

Reply:
`shift_selection_check_pre` execute a `deselect()` when you pass a false value.

Then the ` if (selection.enabled)` condition is never executed:

```
if ((caret_column < selection.begin) || (caret_column > selection.end) || !selection.enabled) {
	deselect();
	selection.start_column = caret_column;
	selection.creating = true;
} else if (selection.enabled) {
	selection.drag_attempt = true;
}
```